packages/airnode-protocol/contracts/rrp/TemplateUtils.sol

```diff
@@ -62,10 +62,11 @@ contract TemplateUtils is ITemplateUtils {
             bytes[] memory parameters
         )
     {
-        airnodes = new address[](templateIds.length);
-        endpointIds = new bytes32[](templateIds.length);
-        parameters = new bytes[](templateIds.length);
-        for (uint256 ind = 0; ind < templateIds.length; ind++) {
+        uint256 templateIdsLength = templateIds.length;
```

Comment:
It's surprising to me that the solidity compiler wouldn't automatically optimise out something like this - crazy.

Reply:
I reproduced this locally and amazingly caching a bytes32 value from a mapping saved around 3407 gas units for a 0 optimisation runs value and 1743 gas units for a 200 optimisation runs value. For a 2000 optimisation runs value the cost delta increased to 1788 - so excessive optimiser runs isn't just diminishing returns, it can in fact become more expensive.